docs/release-notes/self-hosted/self-hosted-v12.0.0.md

```diff
@@ -0,0 +1,106 @@
+---
+rss_title: Codacy release notes RSS feed
+rss_href: /feed_rss_created.xml
+description: Release notes for Codacy Self-hosted v12.0.0.
+codacy_tools_version_old: https://github.com/codacy/codacy-tools-sh/releases/tag/sh-1.1.6
+codacy_tools_version_new: https://github.com/codacy/codacy-tools-sh/releases/tag/sh-1.2.1
+---
+
+# Self-hosted v12.0.0
+
+These release notes are for [Codacy Self-hosted v12.0.0](https://github.com/codacy/chart/releases/tag/12.0.0){: target="_blank"}, released on June 21, 2023.<!-- TODO Update release date -->
+
+📢 [Visit the Codacy roadmap](https://roadmap.codacy.com) and <span class="skip-vale">let us know</span> your feedback on both new and planned product updates!
+
+<!--TODO Check these issues manually
+Jira issues without release notes
+-   https://codacy.atlassian.net/browse/TS-110 TODO review
+
+Jira issues with disabled release notes
+-   https://codacy.atlassian.net/browse/TS-369 TODO Notify Paulo that we're ignoring this one: https://codacy.atlassian.net/browse/TS-369?focusedCommentId=56541
+-   https://codacy.atlassian.net/browse/TS-356 TODO we may want to notify this: ESlint: install plugin vue-scoped-css
+-->
+
+## Upgrading Codacy Self-hosted
+
+Follow the steps below to upgrade to Codacy Self-hosted v12.0.0:
+
+1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v12.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the following versions:<!--TODO Update CLI tool versions-->
+
+    -   [Codacy Analysis CLI MAJOR.MINOR.PATCH](https://github.com/codacy/codacy-analysis-cli/releases/tag/MAJOR.MINOR.PATCH)
+    -   [Codacy Coverage Reporter MAJOR.MINOR.PATCH](https://github.com/codacy/codacy-coverage-reporter/releases/tag/MAJOR.MINOR.PATCH)
+
+## Breaking changes
+
+If you're updating Kubernetes to v1.25, upgrade to Codacy Self-hosted v12.0.0 before doing so, as some Kubernetes endpoints have been deprecated and may lead to service outage if used with Codacy Self-hosted v11.0.0.
+
+If using the monitoring functionality [Grafana + Prometheus + Loki](https://docs.codacy.com/chart/configuration/monitoring/#setting-up-monitoring-using-grafana-prometheus-and-loki), ensure Pod Security Policies are disabled for Loki and Promtail by running the following commands after upgrading to Codacy Self-hosted v12.0.0 but before updating Kubernetes to v1.25:
```

Comment:
@tfrgoncalves

We may need to add this command to the [relevant upgrade page](https://docs.codacy.com/chart/maintenance/upgrade/).

Could you confirm? Would updating point 4 of that page be enough?

Reply:
What happens is that, if a user has the Grafana + ... installation in use and ran those helm commands we provide they need to disable those policies by running the helm update for them before upgrading to kubernetes 1.25. So, it is a "before upgrading to Kubernetes X" requirement. It may be more clear for users to have an approach similar to 'In particular, Codacy Self-hosted v5.0.0...' by the beginning of the page. @lolgab does it also make sense for you?

Reply:
Thanks @tfrgoncalves! So, to clarify (keep in mind I have zero context on these procedures, so I'm guesstimating step 2):

1. [Upgrade Codacy](https://docs.codacy.com/chart/maintenance/upgrade/) (you mention "preferably", but it's best to give clear steps here)
2. Run a `helm update` on Loki and Promtail, essentially using the instructions we already provide ([Loki](https://docs.codacy.com/chart/configuration/monitoring/#2-installing-loki), [Promtail](https://docs.codacy.com/chart/configuration/monitoring/#3-installing-promtail))
  * When updating, customize [values-loki.yaml](https://docs.codacy.com/chart/values-files/values-loki.yaml) and [values-promtail.yaml](https://docs.codacy.com/chart/values-files/values-promtail.yaml) to include the option `podSecurityPolicy: enabled: false` (formatted as yaml)
  * Note: if the above holds true, I suggest updating those files for the v12 docs
3. Finally, update to Kubernetes 1.25

Makes sense?

Reply:
All making sense. Upgrading Codacy to this version must always happen before upgrading Kubernetes to 1.25 indeed, not only "preferably".

Reply:
Thanks!

While updating `values-loki.yaml` and `values-promtail.yaml`, I noticed that there's a line already setting the PSP value to false. I suspect that the syntax of the option `podSecurityPolicy` that I mentioned above may be incorrect, since it was suggested by ChatGPT. The one below is likely the correct one. I'd simply add a comment to the relevant line to clarify when not to change its value:

```
rbac:
  pspEnabled: false # must be set to `false` with Kubernetes 1.25 and higher
```

I'd then simplify the note to read something like this, plus an admonition similar to 'In particular, Codacy Self-hosted v5.0.0...' on the upgrade page, as you suggest.

```suggestion
If using the monitoring functionality [Grafana + Prometheus + Loki](https://docs.codacy.com/chart/configuration/monitoring/#setting-up-monitoring-using-grafana-prometheus-and-loki), ensure Pod Security Policies are disabled for Loki and Promtail by following the relevant update instructions ([Loki](https://docs.codacy.com/chart/configuration/monitoring/#2-installing-loki), [Promtail](https://docs.codacy.com/chart/configuration/monitoring/#3-installing-promtail)) after upgrading to Codacy Self-hosted v12.0.0 but before updating Kubernetes to v1.25.
```

Reply:
The idea is for users to run helm upgrade against our most recent x-values.yaml, as seen in these two changes ([here](https://github.com/codacy/chart/commit/e246a14770aad096d1d1d8c85db018396e2ec079) and [here](https://github.com/codacy/chart/commit/c3423b632ffc2ceb165074d576c09ef9f37c6aec)) I made some time ago.

There is also the case users have not set up the [Grafana + Prometheus + Loki] installation in accordance to our values, which may lead to issues on those deployments, but that is out of our concern to solve.

Reply:
Had overseen the `podSecurityPolicy: enabled: false` error part, sorry.

Reply:
Thanks for clarifying. I'll update this line and avoid too much detail on the yaml files.

Reply:
Long story short, they just need to rerun the helm upgrade if they used our values.yaml files or make sure their deployments disable rbac otherwise, before the upgrade to kubernetes 1.25.